arangod/CMakeLists.txt

```diff
@@ -833,6 +833,10 @@ add_library(arango_rocksdb STATIC
   ${ADDITIONAL_LIB_ARANGO_ROCKSDB_SOURCES}
 )
 
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+  target_compile_options(arango_rocksdb PUBLIC "-maes")
```

Comment:
This will allow usage of AES instructions for all code in this library. I doubt the compiler will emit AES instructions for other code parts than the actual encryption, but it is allowed to do so.

I think the minimum required target platform for running our packages is Westmere, which should include all the AES instructions if I am not wrong. So this would be safe. But it would be good if someone else could confirm this!

Reply:
so I was not able to just add the compile option on the one relevant file. I think the compiler will never output these instructions accidentally, so it should be ok

Reply:
I'd guess the same, though I didn't find any explicit documentation. I'm fine with this. But if you want, you can add another library `arango_rocksdb_aes` or so, and add the option only to that target.

Reply:
@goedderz: sounds like a good idea to me!